src/core/drive/page_renderer.ts

```diff
@@ -1,19 +1,26 @@
 import { Renderer } from "../renderer"
 import { PageSnapshot } from "./page_snapshot"
 import { ReloadReason } from "../native/browser_adapter"
-import { activateScriptElement, waitForLoad } from "../../util"
+import { activateScriptElement, waitForLoad, nextEventLoopTick, getBodyElementId } from "../../util"
 
 export class PageRenderer extends Renderer<HTMLBodyElement, PageSnapshot> {
-  static renderElement(currentElement: HTMLBodyElement, newElement: HTMLBodyElement) {
+  static async renderElement(currentElement: HTMLBodyElement, newElement: HTMLBodyElement) {
+    await nextEventLoopTick()
+
     if (document.body && newElement instanceof HTMLBodyElement) {
-      document.body.replaceWith(newElement)
+      const bodyElementId = getBodyElementId()
+
+      const currentBody = (bodyElementId && document.querySelector(`#${bodyElementId}`)) || document.body
```

Comment:
@agrobbin Is there a reason why we are using `document` here instead of `currentElement`?

Reply:
@marcoroth I wish I could tell you. I followed the existing implementation, which used `document.body` rather than `currentElement`, for reasons I'm not aware of.